ansible_collections/arista/avd/roles/eos_designs/templates/facts/switch/switch.j2

```diff
@@ -348,7 +350,7 @@ switch:
 {%             set add_svis = [] %}
 {%             for svi in tenants[tenant].vrfs[vrf].svis | arista.avd.natural_sort %}
 {%                 for svi_tag in tenants[tenant].vrfs[vrf].svis[svi].tags | arista.avd.natural_sort %}
-{%                     if svi_tag in switch_filter_tags or svi_tag == switch_data.group or "all" in switch_filter_tags %}
+{%                     if svi_tag in switch_filter_tags or svi_tag == switch_data.group | arista.avd.default('group_not_set') or "all" in switch_filter_tags %}
```

Comment:
Looks good. You can try ```switch_data.group is arista.avd.defined(svi_tag)``` instead of the ```==```. Let me know if the suggestion looks good!

Reply:
Very good suggestion. Thanks